charts/opentelemetry-collector/templates/service.yaml

```diff
@@ -1,4 +1,3 @@
-{{- if or (eq .Values.mode "deployment") (eq .Values.mode "statefulset") -}}
```

Comment:
Instead of removing this restriction, let's add ingress.enabled to the or list. I don't think we want to start adding services for daemonset users who aren't using ingress options.